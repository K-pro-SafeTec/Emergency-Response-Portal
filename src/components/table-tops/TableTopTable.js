import React from 'react';
import Table from '@material-ui/core/Table';
import TableHead from '@material-ui/core/TableHead';
import TableBody from '@material-ui/core/TableBody';
import TableRow from '@material-ui/core/TableRow';
import TableCell from '@material-ui/core/TableCell';
import LinkTableRow from '../shared/LinkTableRow';
import { tableTopScenarioById } from '../../dummy-data/tableTopScenario';
import {getDateFormatYMD} from "../../helpers/calendar-helper";


const TableTopTable = ({ instances }) => (
  <Table component="div">
    <TableHead component="div">
      <TableRow component="div">
        <TableCell component="div">Dato</TableCell>
        <TableCell component="div">DFU</TableCell>
        <TableCell component="div">Tittel</TableCell>
        <TableCell component="div">Scenario</TableCell>
      </TableRow>
    </TableHead>
    <TableBody component="div">
      {instances
        .map(instance => [tableTopScenarioById[instance.scenario], instance])
        .map(([scenario, instance]) => (
          <LinkTableRow
            key={`${instance.scenario}/${getDateFormatYMD(instance.start)}`}
            to={`../../scenarios/${instance.scenario}/${getDateFormatYMD(instance.start)}/`}
          >
            <TableCell component="div">
              {getDateFormatYMD(instance.start)}
            </TableCell>
            <TableCell component="div">
              {scenario.dfuNr}
            </TableCell>
            <TableCell component="div">
              {scenario.dfuTitle}
            </TableCell>
            <TableCell component="div">
              {scenario.scenario}
            </TableCell>
          </LinkTableRow>
        ))
      }
    </TableBody>
  </Table>
);

export default TableTopTable;
