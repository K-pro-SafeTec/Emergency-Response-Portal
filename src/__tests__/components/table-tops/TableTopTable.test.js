import React from 'react';
import TestRenderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom'
import TableTopTable from "../../../components/table-tops/TableTopTable";

let testRenderer;
let testInstance;

beforeAll(() => {
  testRenderer = TestRenderer.create(
    <Router>
      <TableTopTable
        instances={[{
          scenario: 4,
          start: new Date(2018, 8, 24),
        },
        {
          scenario: 4,
          start: new Date(2018, 8, 8),
        },
        {
          scenario: 4,
          start: new Date(2018, 8, 22),
        },
        {
          scenario: 3,
          start: new Date(2018, 9, 5),
        },
        ]}
      />
    </Router>
  );
  testInstance = testRenderer.root;
});

test('renders correctly', () => {
  const tree = testRenderer.toJSON();
  expect(tree).toMatchSnapshot();
});
