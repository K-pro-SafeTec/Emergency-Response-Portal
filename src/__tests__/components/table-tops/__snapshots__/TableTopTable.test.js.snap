// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiTable-root-1"
>
  <div
    className="MuiTableHead-root-2"
  >
    <div
      className="MuiTableRow-root-3 MuiTableRow-head-6"
    >
      <div
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-head-9"
        scope="col"
      >
        Dato
      </div>
      <div
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-head-9"
        scope="col"
      >
        DFU
      </div>
      <div
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-head-9"
        scope="col"
      >
        Tittel
      </div>
      <div
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-head-9"
        scope="col"
      >
        Scenario
      </div>
    </div>
  </div>
  <div
    className="MuiTableBody-root-16"
  >
    <a
      className="MuiTableRow-root-3 MuiTableRow-hover-5 LinkTableRow-noUnderline-17"
      href="/scenarios/4/2018-09-24/"
      onClick={[Function]}
    >
      <div
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10"
      >
        2018-09-24
      </div>
      <div
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10"
      >
        4
      </div>
      <div
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10"
      >
        Fare for kollisjon
      </div>
      <div
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10"
      >
        Drivende fartøy på kollisjonskurs
      </div>
    </a>
    <a
      className="MuiTableRow-root-3 MuiTableRow-hover-5 LinkTableRow-noUnderline-17"
      href="/scenarios/4/2018-09-08/"
      onClick={[Function]}
    >
      <div
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10"
      >
        2018-09-08
      </div>
      <div
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10"
      >
        4
      </div>
      <div
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10"
      >
        Fare for kollisjon
      </div>
      <div
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10"
      >
        Drivende fartøy på kollisjonskurs
      </div>
    </a>
    <a
      className="MuiTableRow-root-3 MuiTableRow-hover-5 LinkTableRow-noUnderline-17"
      href="/scenarios/4/2018-09-22/"
      onClick={[Function]}
    >
      <div
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10"
      >
        2018-09-22
      </div>
      <div
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10"
      >
        4
      </div>
      <div
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10"
      >
        Fare for kollisjon
      </div>
      <div
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10"
      >
        Drivende fartøy på kollisjonskurs
      </div>
    </a>
    <a
      className="MuiTableRow-root-3 MuiTableRow-hover-5 LinkTableRow-noUnderline-17"
      href="/scenarios/3/2018-10-05/"
      onClick={[Function]}
    >
      <div
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10"
      >
        2018-10-05
      </div>
      <div
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10"
      >
        3
      </div>
      <div
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10"
      >
        Akutt utslipp
      </div>
      <div
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10"
      >
        Oljeutslipp til sjø fra innretninig
      </div>
    </a>
  </div>
</div>
`;
